Added Options to Output Sprite Sheet, Merge Duplicates and Trim Cels

Bastani opened the pull request:
This PR adds a few new options: 

- The Aseprite Texture importer now allows you to export as a sprite sheet.

I particularly have multiple aseprite files that I'd like to swap which have the same animations. This allows us to setup the animations with the inspector dock and then swap the texture programatically through resources.

- The Aseprite Texture (Split By Layer) importer now allows you to Trim Cels and Merge Duplicate Layers

I also like to design my UI in Aseprite and tend to use the same texture for the same layer, but I don't necessarily want it to output those duplicate layers, nor do I want it to leave them untrimmed, so this fixes that problem as well.

<img width="404" alt="image" src="https://github.com/user-attachments/assets/dd930b01-7ac0-4f29-8eb1-2d2157a1c73b" />
<img width="1304" alt="image" src="https://github.com/user-attachments/assets/440ec875-237b-4c80-b35a-23c8fdbedeb7" />
<img width="405" alt="image" src="https://github.com/user-attachments/assets/79a4fc50-5a17-4b1a-a48d-9c605c3bd3fb" />
<img width="317" alt="image" src="https://github.com/user-attachments/assets/1d23ee9d-e381-426b-82fb-9636287dacdf" />


<details>

<summary>Old PR</summary>

This PR adds a new import option to create a custom static texture that allows you to select the sheet type, sheet columns, and fps amount. 

I particularly have multiple aseprite files that I'd like to swap which have the same animations. This allows us to setup the animations with the inspector dock and then swap the texture programatically through resources.

<img width="352" alt="image" src="https://github.com/user-attachments/assets/c9b2737d-6719-4525-9a8a-e61cf89cf6ff" />

</details>

Bastani committed:
Added Aseprite Texture Custom

viniciusgerevini commented:
Thanks @Bastani . There are a few issues. This is missing the export priority configuration. Also, the FPS config is not used anywhere for the spritesheet, so it should not be included. 

I reckon it makes more sense to add sheet options to the existing texture importer rather than creating a new one. The only difference I see between the two importers is the `first_frame_only` config, which could be exposed as an importer option.

Bastani committed:
Added new trim option

Bastani committed:
Cleanup

Bastani committed:
Added option to merge duplicate layers

Bastani commented:
A lot has changed, I've added a new list_layers_json method which can simulate the merge duplicates option in aseprite.
I was playing around with plugins a bit but I merged them together once I figured out how they fully work.

I'm not happy with how disabling and enabling First Frame Only option doesn't act immediately for the _get_option_visibility, but it will have to do for now.

Hope this should be good enough to merge now :)

viniciusgerevini commented on addons/AsepriteWizard/importers/static_texture_multiple_import_plugin.gd:
```diff
@@ -18,6 +18,8 @@ func _get_import_options(_path, _i):
 	return [
 		{"name": "layer/exclude_layers_pattern", "default_value": config.get_default_exclusion_pattern()},
 		{"name": "layer/only_visible_layers",    "default_value": false},
+		{"name": "layer/trim_cels",    			 "default_value": false},
```
Trimming is not a layer option, but a sheet. Could you rename this prop to be `sheet/trim_cels` so it's grouped correctly?

viniciusgerevini commented on addons/AsepriteWizard/importers/static_texture_import_plugin.gd:
```diff
@@ -27,8 +27,24 @@ func _get_import_options(_path, _i):
+	if ((option == "sheet/sheet_type" or option == "sheet/sheet_columns") and 
+		options.has("first_frame_only")):
+		return options["first_frame_only"] == false
+	return true
```
Yeah, the visibility issue you mentioned is an open bug in Godot. I rather not have this logic because it makes prop discoverability harder. I've seen people confused about this because they only show up if you deselect and re-select the file.

viniciusgerevini commented:
@Bastani thanks for putting the time to improve the plugin. I made some minor comments that I think should be ironed out before the release. But I'm also happy to do it myself after merge if you prefer.

The "merge duplicate layers" is an interesting feature. I'm happy to bring it in, I just want to point out that for your UI example, I think this is a "fragile" feature similar to importing using "only visible layers".

For example, from what I've seen, when Aseprite find duplicated layers, it only returns the last one. So imagine a scenario where you have Layer 1 and Layer 2 that are different textures. If you are using Layer 1, with merge duplicates enabled, and at some point they become the same texture, the Layer 1 file will be removed in Godot without any previous warning. The same is true if "only visible layers" is enabled, so this is not a new issue, I'm just pointing out why this can be a problem when you are referencing an specific file directly.

The option is still quite nice for other scenarios where the folder is scanned though. Thanks

Bastani committed:
Addressed issues

Bastani commented:
@viniciusgerevini 
> The "merge duplicate layers" is an interesting feature. I'm happy to bring it in, I just want to point out that for your UI example, I think this is a "fragile" feature similar to importing using "only visible layers".

Took me a minute to wrap my head around what you meant. I think I also noticed that Godot would instead store the texture locally in the scene in a similar situation. I had to re-do all my references to fix it but luckily I noticed it in git before I commited it.

I've also addressed the comments, glad to contribute :)

viniciusgerevini commented:
Thank you :rocket:
